src/main/java/use_case/sub_menu/SubViewInteractor.java

```diff
@@ -0,0 +1,22 @@
+package use_case.sub_menu;
+
+import interface_adapter.sub_menu.SubViewPresenter;
+import use_case.signup.SignupOutputBoundary;
+
+import java.util.Objects;
+
+public class SubViewInteractor implements SubViewInputBoundary{
+    final SubViewOutputBoundary userPresenter;
+
+    public SubViewInteractor(SubViewOutputBoundary subViewOutputBoundary) {
+        this.userPresenter = subViewOutputBoundary;
+    }
+
+    public void execute(SubViewInputData subViewInputData) {
+        if (Objects.equals(subViewInputData.getNext_panel(), "search")) {
+            userPresenter.prepareSearchView();
+        } else if (Objects.equals(subViewInputData.getNext_panel(), "recommendation")) {
+            userPresenter.prepareApplyView();
+        }
```

Comment:
Maybe throw an error if SubViewInputData next pane is not one of the expected panes